### ISSUE TITLE
about dataset split

### ISSUE THREAD

**Issue by CrownX** [piotr-teterwak/erm_plusplus#4]
according dataset.py file, if args.train_val_split = 0.8, train_split is from source domain (0.8 length) but test_split is from target domain (0.2 length)

In my understanding, during the training process, DG assumes we do not know the target domain.
How to explain your setting?

**Comment by piotr-teterwak** [comment]
Hi @CrownX , you are correct that in DG the assumption is that we do not have access to the target domain. However, validation is still useful for searching for the correct training length. In our work, we do two phases of training. In the first, we use a validation split **of the source domains, not the target domains** to find the correct number of steps to train for. In the second stage we set `args.train_val_split = -1` and train for the number of steps found in the first stage, in order to take advantage of the full data.

**Comment by CrownX** [comment]
From dataset.py, _constrcut_dataset_helper(), 'datasets_split_train' is generated based on  train_datasets (from args.training_data) and datasets_split_val is based on val_datasets (from args.validation_data).

From the readme.md, the input for domainnet dataset is '--training_data "clipart infograph real quickdraw sketch" --validation_data painting'.

Obviously, thie training_data and validation_data is different, not as your description. 'a validation split of the source domains'. Could you check this?

**Comment by piotr-teterwak** [comment]
Good catch! I must have gotten sloppy making the README.md. I fixed it and double checked my launcher scripts, and it is as I described and not as it was in the README.md. Let me know if that answers your question.

**Comment by CrownX** [comment]
Have got it. Thanks for your quick response!
